File: hackerrank/algorithms/dilshad/3d-surface-area.js
function surfaceArea(A) {

    let arrLen = A.length;
    let len = A[0].length;
    let left = [];
    let right = [];
    let up = [];
    let down = [];
    let temp = [];
    let adjacent = [];
    let x, y;
    let total = 0;
for(let i = 0; i< arrLen; i++){
    for(let j = 0; j < len; j++) {
        left = [i, j-1];
        right = [i, j+1];
        up = [i-1, j];
        down = [i+1, j];
        temp = [left, right, up, down]
        adjacent = [];
        console.log(temp);
        for(let k = 0; k< temp.length; k++)
        {
            x = temp[k][0];
            y = temp[k][1];

            if((x >= 0 && x < arrLen) && (y >= 0 && y < len))
            {
                adjacent.push(A[x][y]);
            }

        }
        let height = A[i][j];
        let area_threshold = ((4-adjacent.length)*height) + 2;

        let temp_sum = 0;

        for (let k = 0; k < adjacent.length; k++)
        {
            if (height > adjacent[k])
            {
                temp_sum += (height - adjacent[k]);
            }
        }

        total += (temp_sum + area_threshold)

    }
}
return total;
}

File: hackerrank/algorithms/gaurav/3d_surface_area.py
# https://www.hackerrank.com/challenges/3d-surface-area/problem

#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the surfaceArea function below.
def surfaceArea(A):
    H = len(A)
    W = len(A[0])
    total = 0
    for i in range(H):
        for j in range(W):

            left = (i, j-1)
            right = (i, j+1)
            up = (i-1, j)
            down = (i+1, j)

            temp = [left, right, up, down]
            adjacent = list()
            
            for cord in temp:
                x = cord[0]
                y = cord[1]
                if (x >= 0 and x < H) and (y >= 0 and y < W):
                    adjacent.append(cord)
                else:
                    pass
            
            height = A[i][j]
            
            adj_heights = list()

            for block in adjacent:
                adj_heights.append(A[block[0]][block[1]])

            area_threshold = ((4 - len(adjacent)) * height) + 2
            lst = list()

            for block_height in adj_heights:
                if height > block_height:
                    lst.append(height - block_height)
                else:
                    pass
            
            block_area = sum(lst) + area_threshold

            total += block_area
    
    return(total)


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    HW = input().split()

    H = int(HW[0])

    W = int(HW[1])

    A = []

    for _ in range(H):
        A.append(list(map(int, input().rstrip().split())))

    result = surfaceArea(A)

    fptr.write(str(result) + '\n')

    fptr.close()
